test/ketu/spec_test.clj

```diff
@@ -0,0 +1,134 @@
+    (tc/quick-check
+      10
+      (prop/for-all [name (gen/such-that (complement clojure.string/blank?) gen/string)]
+        (let [sink-opts {:ketu/name name}]
+          (is (s/valid? :ketu/sink-opts sink-opts))))))
+
+  (testing "that passing all possible valid fields is valid"
+    (tc/quick-check
+      50
+      (prop/for-all [name                      (gen/such-that (complement clojure.string/blank?) gen/string)
+                     topic                     (gen/such-that (complement clojure.string/blank?) gen/string)
+                     shape                     (gen/elements [:value
+                                                              :key-value-vector
+                                                              :key-value-map
+                                                              [:vector :key :value :topic :partition :timestamp :headers]
+                                                              [:map :key :value :topic :partition :timestamp :headers]])
+                     sender-threads-num        (gen/large-integer* {:min 1})
+                     sender-threads-timeout-ms (gen/large-integer* {:min 0})
+                     close-producer?           gen/boolean
+                     producer-close-timeout-ms (gen/large-integer* {:min 1})
+                     callback                  (gen/elements [boolean? int? contains? neg?])
+                     callback-obj              (gen/return (reify Callback
+                                                             (onCompletion [_ _ _])))
+                     create-callback           (gen/elements [boolean? int? contains? neg?])
+                     create-callback-obj       (gen/elements [boolean? int? contains? neg?])]
+        (let [sink-opts {:ketu/name                           name
+                         :ketu/topic                          topic
+                         :ketu.sink/shape                     shape
+                         :ketu.sink/sender-threads-num        sender-threads-num
+                         :ketu.sink/sender-threads-timeout-ms sender-threads-timeout-ms
+                         :ketu.sink/close-producer?           close-producer?
+                         :ketu.sink/producer-close-timeout-ms producer-close-timeout-ms
+                         :ketu.sink/callback                  callback
+                         :ketu.sink/callback-obj              callback-obj
+                         :ketu.sink/create-callback           create-callback
+                         :ketu.sink/create-callback-obj       create-callback-obj}]
+          (is (s/valid? :ketu/sink-opts sink-opts))))))
+
+  (testing "that passing valid required fields and invalid optional fields is invalid"
+    (tc/quick-check
+      50
+      (prop/for-all [name                      (gen/such-that (complement clojure.string/blank?) gen/string)
+                     topic                     gen/large-integer
```

Comment:
I'm not sure it's possible, but it would be nice to have more interesting invalid values in this test. For example, a blank string `topic`, a truthy non-boolean `close-producer?`, a negative `sender-threads-num` etc. As it is now, isn't it generating many strings which are obviously invalid?

Reply:
Added suggested generators to the tests